src/main/java/frc/robot/subsystems/pivot/factories/PivotFactory.java

```diff
@@ -0,0 +1,46 @@
+package frc.robot.subsystems.pivot.factories;
+
+import com.ctre.phoenix6.controls.PositionVoltage;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.Robot;
+import frc.robot.constants.GlobalConstants;
+import frc.robot.constants.IDs;
+import frc.robot.hardware.motor.phoenix6.TalonFXWrapper;
+import frc.robot.hardware.request.phoenix6.Phoenix6AngleRequest;
+import frc.robot.hardware.signal.phoenix.Phoenix6SignalBuilder;
+import frc.robot.subsystems.pivot.Pivot;
+import frc.utils.AngleUnit;
+
+public class PivotFactory {
+
+	public static Pivot generatePivot(String logPath) {
+		return switch (Robot.ROBOT_TYPE) {
+			case REAL -> getRealPivot(logPath);
+			default -> null;
+		};
+	}
+
+	private static Pivot getRealPivot(String logPath) {
+		TalonFXWrapper motorWrapper = new TalonFXWrapper(IDs.TalonFXs.PIVOT_MOTOR_ID);
+		motorWrapper.applyConfiguration(PivotRealConstants.TALON_FX_CONFIGURATION);
+
+		return new Pivot(
+			logPath,
+			PivotRealConstants.getTalonFXMotor(logPath, motorWrapper),
+			new Phoenix6AngleRequest(new PositionVoltage(0).withEnableFOC(true)),
+			Phoenix6SignalBuilder.generatePhoenix6Signal(motorWrapper.getPosition(), 100, AngleUnit.ROTATIONS),
+			Phoenix6SignalBuilder.generatePhoenix6Signal(motorWrapper.getPosition(), GlobalConstants.CANIVORE_UPDATE_FREQUENCY),
+			Phoenix6SignalBuilder.generatePhoenix6Signal(motorWrapper.getVelocity(), GlobalConstants.CANIVORE_UPDATE_FREQUENCY),
+			Phoenix6SignalBuilder.generatePhoenix6Signal(motorWrapper.getMotorVoltage(), GlobalConstants.CANIVORE_UPDATE_FREQUENCY),
+			Phoenix6SignalBuilder.generatePhoenix6Signal(motorWrapper.getStatorCurrent(), GlobalConstants.CANIVORE_UPDATE_FREQUENCY)
+		);
+	}
```

Comment:
in intake i did this in RealIntakeConstants, we should decide on a way. same with IntakeStuff/PivotStuff